### PR TITLE
VACMS-9714 script change phone to 988

### PR DESCRIPTION
**Pull request by omahane**
## Description

Closes to #9714 

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As user _uid_ with _user_role_
1. Do this
   - [ ] Validate that
2. Then
   - [ ] Validate that
3. Then validate Acceptance Criteria from issue
   - [ ] a
   - [ ] b
   - [ ] c

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by ndouglas** [comment]
I'm seeing `138` when running the script and `138` when I do 
```sql
SELECT entity_id FROM node__field_operating_status_emerg_inf AS patient_resources WHERE patient_resources.field_operating_status_emerg_inf_value REGEXP '800\-273\-8255';
```

So that all seems consistent.  I also see `138` when I do 
```sql
SELECT entity_id FROM node__field_operating_status_emerg_inf AS patient_resources WHERE patient_resources.field_operating_status_emerg_inf_value LIKE '%8255%';
```

So the good news is that I think your regular expression is catching all of the nodes.

I also dumped the `$nids_to_update` variable locally and it matches the results of the SQL query.

So I think you're good -- I think it's just a matter of doing the text replacement.

Is there anything specific you're concerned about moving forward?

File: scripts/content/VACMS-9714-change-crisis-to-988.php [hunk 1]
```diff
@@ -48,69 +56,72 @@ function va_gov_change_crisis_hotline_to_988_nodes(array &$sandbox) {
...
     $sandbox['#finished'] = 1;
     return "No nodes found for processing.\n";
   }
 
   $limit = 25;
 
-//   // Load entities.
+  //   // Load entities.
   $node_ids = array_slice($sandbox['nids_to_update'], 0, $limit, TRUE);
   $nodes = $node_storage->loadMultiple($node_ids);
```
[the remark this answers is not in the document]
Comment (omahane) [reply scripts/content/VACMS-9714-change-crisis-to-988.php:74]: Thanks for taking a look at this. I'll need to test that and see.

File: scripts/content/VACMS-9714-change-crisis-to-988.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  if (!isset($sandbox['total'])) {
+    // Hours field is attached to service_locations in all places,
+    // save for facilities.
+    // Facilities db search returned 0 results for timezones in comments.
+
+    $paragraph_query = Drupal::database()->select('node__field_operating_status_emerg_inf', 'patient_resources');
+    $paragraph_query->join('node__field_facility_operating_status', 'nffos', 'patient_resources.entity_id = nffos.entity_id');
+    $paragraph_query->fields('patient_resources', ['entity_id']);
+    $paragraph_query->groupBy('entity_id');
+    $paragraph_query->condition('patient_resources.field_operating_status_emerg_inf_value', $pattern_string, 'REGEXP');
```
Comment (ndouglas) [inline scripts/content/VACMS-9714-change-crisis-to-988.php:48]: I (perhaps incorrectly) translate this to:

```SQL
SELECT patient_resources.entity_id FROM node__field_operating_status_emerg_inf AS patient_resources JOIN node__field_facility_operating_status AS nffos ON patient_resources.entity_id = nffos.entity_id WHERE patient_resources.field_operating_status_emerg_inf_value REGEXP '(\<a.*\>)?(1[\-\.])?800[\-\.]273[\-\.]8255(\<\/a\>)?';
```

I get 1166 rows from this, but they're mostly duplicates.  If I do SELECT DISTINCT, I only get 135 rows.  I haven't poked around much, but this makes me think there might be an issue where you're saving a node repeatedly, changing one field's instances of the search string, and then you check the node and still see occurrences.

I'm in Retro, so I can't dive further yet.

**Comment by omahane** [comment]
Tested VACMS-9714-change-crisis-to-988-VAMC-detail-field-description.php
Found 140 instances
Screen Shot 2022-07-18 at 9 21 31 AM

**Comment by omahane** [comment]
Tested VACMS-9714-change-crisis-to-988.php
Found 138 instances.
Had to run it twice, as one (node/18649) still had the old number after the first run.
Screen Shot 2022-07-18 at 1 02 26 PM

File: scripts/content/VACMS-9714-change-crisis-to-988.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (swirtSJW) [reply scripts/content/VACMS-9714-change-crisis-to-988.php:68]: Nate is spot on with this.   Jay has a solution to loading node, then checking to see if the most recent revision is the default, and if it is not, then updates both the default, and creates a new revision and moves it up the revision list so editors don't get confused AND don't wipe out the change when they publish their draft.

**Comment by swirtSJW** [comment]
Jay is also adding some of these data tools to a library that Nate created a while back so that we don't have to keep reinventing some of these patterns.

**Pull request closed without merging**